### DISCUSSION TITLE
Homebrew: `emsdk` not found (`brew install emscripten`)

### DISCUSSION THREAD

**Discussion by michaelweinold** [emscripten-core/emscripten#19143] [Q&A] [answered]
While [the documentation makes it clear](https://emscripten.org/docs/getting_started/downloads.html) that the only officially supported way to install the `emsdk` is to clone the git repo and install from there, [homebrew's `brew install emscripten`](https://formulae.brew.sh/formula/emscripten) is mentioned on the same page.

After installing emscripten using homebrew, I can use the `emcc`, but not the `emsdk`:

```
(base) michaelweinold@Michaels-MacBook-Pro ~ % emcc --check
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.35-git
shared:INFO: (Emscripten: Running sanity checks)
```

```
(base) michaelweinold@Michaels-MacBook-Pro ~ % emsdk
zsh: command not found: emsdk
```

 I could not find any information on whether the homebrew command actually installs the SDK, or how to enable it ([unanswered question on StackOverflow](https://stackoverflow.com/q/73452405/).

**Comment by sbc100** [comment]
Installing via brew is an alternative to install via emsdk, so if you install via brew I wouldn't expect to find emsdk in your PATH.

If `emcc --check` then you should be good to go.

**Reply by sbc100** [reply]
If you want to use emsdk instead of the homebrew package then you would need to `git clone` the emsdk reprository as stated in the docs you refer to: https://emscripten.org/docs/getting_started/downloads.html#installation-instructions-using-the-emsdk-recommended

**Reply by michaelweinold** [reply]
Ok, I see. I had hoped homebrew would offer a quick alternative to the multi-step installation process of emsdk (cloning repo, installing, activating, adding to `PATH`).

**Reply by michaelweinold** [reply]
`emcc --check` works fine following `brew install emscripten`:

```
(base) michaelweinold@Michaels-MacBook-Pro ~ % emcc --check
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.35-git
```

**Reply by sbc100** [reply]
> Ok, I see. I had hoped homebrew would offer a quick alternative to the multi-step installation process of emsdk (cloning repo, installing, activating, adding to `PATH`).

I mean that kind of is exactly what it does.  Its basically another way to get the same tools that emsdk would give you, but its supported by the homebrew folks rather than directly by the emscripten team.

**Reply by michaelweinold** [reply]
Does it provide the same tools that emsdk would give me? I am trying to point to the emsdk directory as per [the build instructions of the emscripten-forge project](https://emscripten-forge.readthedocs.io/en/latest/src/projects/recipes/local_build.html) (cf. _create a .emsdkdir file_), but the homebrew installation under 

```
/opt/homebrew/Cellar/emscripten/3.1.35/bin
```

does not seem to contain an emsdk sub-directory (or the appropriate files)?

**Reply by sbc100** [reply, marked as answer]
I believe the homebrew package provides all the same tools that emsdk does (i.e. llvm, binaryen, and emscripten), but it uses its own directory layout that matches what homebrew users expect.   If you have more questions about the specifics of the homebrew package then I think that maintainer of the package is probably the best person to ask about it.

If emscripten-forge is specifically saying that it requires emsdk (and not just a working copy of emscripten and its dependencies) than you best bet is likely to simply use emsdk directly and not try to use the homebrew package (which is an alternative to using emsdk).
